python/triton/runtime/cache.py

```diff
@@ -115,7 +115,8 @@ def put(self, data, filename, binary=True) -> str:
         binary = isinstance(data, bytes)
         if not binary:
             data = str(data)
-        assert self.lock_path is not None
```

Comment:
The main problem is that assertions may not be executed in optimized code, so they are more recommended to be used in tests.

Reply:
I don't understand the comment, this is not test code?

Reply:
In my head the canonical use should be:
- `assert` -> this is true by construction and does not depend on user input, this fails only if there is a bug in the compiler
- `raise` -> this is validating user input or expected system state, is expected to sometimes fail even if there are no bugs 

Some of the examples here seem to be perfectly fine as assertions. Agreed that we might not want this tool.

Reply:
> I don't understand the comment, this is not test code?

I meant test code like pytest tests (or regression tests or something else).

Here it seems that there are checks in runtime that can tell the user what is wrong (or maybe not to the user himself, but at least the stack trace will be closer to the problem area and it will be easier to analyze their stack traces in github issues for Triton maintainers), but they may not always be called due to optimizations (ref: https://docs.python.org/3/reference/simple_stmts.html#the-assert-statement), which can lead to more complex errors from a debugging point of view IMO.

> Agreed that we might not want this tool.

Ok, thanks for review! Should I keep any changes or just close the pull request?

Reply:
I think most asserts are used correctly, let's just close the pull request